### PR TITLE
BUG: Limited available colors

### LINKED ISSUE #37967: BUG: Limited available color name list when using style and to_excel
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# Your code here
import pandas as pd

df = pd.DataFrame( {'A': [1, 2, 3],
                   'B': [4, 5, 6],})

# /1 -- It Fails ---
df_out = (df.style.
          highlight_max(color='orangered').
          highlight_min(color='dodgerblue')
         )
df_out.to_excel('pandas_to_excel_color_issue.xlsx')

# /2 -- It works ----
df_out = (df.style.
          highlight_max(color='#FF4500'). # orangered
          highlight_min(color='#1E90FF') # dodgerblue)
         )
df_out.to_excel('pandas_to_excel_color_issue.xlsx')
```

#### Problem description

When using style, the `to_excel` function is not able to handle some common color name that otherwise works with an interactive python console.

See with code block /1 returns : 

```python
c:\program files\python38\lib\site-packages\pandas\io\formats\excel.py:331: CSSWarning: Unhandled color format: 'dodgerblue'
  warnings.warn(f"Unhandled color format: {repr(val)}", CSSWarning)
c:\program files\python38\lib\site-packages\pandas\io\formats\excel.py:331: CSSWarning: Unhandled color format: 'orangered'
  warnings.warn(f"Unhandled color format: {repr(val)}", CSSWarning)
```
And in the Excel file :
image

While the /2 block is fine:
image

#### Expected Output

I understand that the to_excel function only accepts a minimal list of color names (based on the export engine I presume).
It would hence be great if pandas could manage the color Hex conversion under the hood for the user, and hence allow him to use a more large color name list (the one from matplotlib for example with `matplotlib.colors.cnames`), preventing the usage of unreadable color code.
In addition, I don't know if the current Excel output in case of CSS warning is really appropriate (the 'black render'), maybe simply ignoring the formatting would be better? (one could argue that it's more noticeable in the Excel file like that, that's right).

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.8.3.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : fr_FR.cp1252

pandas           : 1.1.4
numpy            : 1.19.3
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.4
setuptools       : 50.3.2
Cython           : None
pytest           : 6.1.2
hypothesis       : None
sphinx           : 3.2.0
blosc            : None
feather          : None
xlsxwriter       : 1.3.2
lxml.etree       : 4.5.2
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.3
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : 1.0.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : 3.6.1
tabulate         : None
xarray           : 0.16.0
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : None

</details>

### PR DESCRIPTION
**Pull request by ma3da**
(GH37967)

- [x] closes #37967
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by ma3da** [comment]
Not sure if bug is the good label...

**Comment by ma3da** [comment]
Two small scripts used for that.
```
from pandas.io.formats.excel import CSSToExcelConverter
from matplotlib.colors import CSS4_COLORS

# check if existing colors match the ones of matplotlib
for name, color in CSSToExcelConverter.NAMED_COLORS.items():
    assert "#" + color == CSS4_COLORS[name]

# generate code for CSSToExcelConverter
for name, color in CSS4_COLORS.items():
    print(f'"{name}": "{color[1:]}",')
```

File: pandas/tests/io/formats/test_to_excel.py [hunk 1]
```diff
@@ -313,3 +314,9 @@ def test_css_to_excel_bad_colors(input_color):
     with tm.assert_produces_warning(CSSWarning):
         convert = CSSToExcelConverter()
         assert expected == convert(css)
+
+
+def tests_css_named_colors_valid():
+    upper_hexs = set(map(str.upper, string.hexdigits))
+    for color in CSSToExcelConverter.NAMED_COLORS.values():
+        assert len(color) == 6 and all(c in upper_hexs for c in color)
```
Comment (ivanovmg) [inline pandas/tests/io/formats/test_to_excel.py:322]: Would you consider adding a test that all colors from ``matplotlib.colors.CS4_COLORS`` are present in ``NAMED_COLORS``?
Or would that be an overkill?

Comment (ma3da) [reply pandas/tests/io/formats/test_to_excel.py:324]: Yeah, I wondered a bit about that, but didn't make my mind.
It's not long to write, so I've pushed a commit for that, but no problem for removing it.

File: pandas/io/formats/excel.py [hunk 2]
```diff
@@ -65,27 +65,156 @@ class CSSToExcelConverter:
         CSS processed by :meth:`__call__`.
     """
 
+    # source: matplotlib 3.3.3, matplotlib.colors.CSS4_COLORS
```
Comment (jreback) [inline pandas/io/formats/excel.py:68]: can we just import these?

**Review by jreback — CHANGES_REQUESTED** [review]
-1 on adding a big list like this. let's just use the mpl machinery here (by importing the color defs)

File: pandas/io/formats/excel.py (hunk 2 above)
Comment (ma3da) [reply pandas/io/formats/excel.py:68]: possible of course, but is it worth introducing a dependency on mpl for excel manipulation?

Comment (ma3da) [reply pandas/io/formats/excel.py]: I used matplotlib nomenclature (_color_data.py), maybe something else will be more adequate

File: pandas/io/formats/_color_data.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+# source: matplotlib._color_data (3.3.3)
```
Comment (jreback) [inline pandas/io/formats/_color_data.py:1]: can you add some more commentary here, why we need this and we don't want to depend on using mpl for this data.

File: pandas/io/formats/excel.py [hunk 4]
```diff
@@ -25,6 +25,8 @@
 from pandas.io.formats.format import get_level_lengths
 from pandas.io.formats.printing import pprint_thing
 
+from ._color_data import CSS4_COLORS
```
Comment (jreback) [inline pandas/io/formats/excel.py:28]: use a fully qualifed import

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note, other enhancements for 1.2

**Comment by jreback** [comment]
cc @charlesdong1991

File: pandas/io/formats/_color_data.py (hunk 3 above)
Comment (ma3da) [reply pandas/io/formats/_color_data.py:5]: done

File: pandas/io/formats/excel.py (hunk 4 above)
Comment (ma3da) [reply pandas/io/formats/excel.py]: done

File: pandas/io/formats/_color_data.py (hunk 3 above)
Comment (jreback) [reply pandas/io/formats/_color_data.py:5]: thanks. Do we have any need for this in plotting itself? @ivanovmg @charlesdong1991 @jorisvandenbossche

**Comment by jorisvandenbossche** [comment]
(removing 1.2 milestone, since this is not critical to include)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ma3da

**Pull request merged**